### PR TITLE
work in tmp dir if --use-tmp-dir (not...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

work in tmp dir if --use-tmp-dir (not required when running as GH action)

Change-Id: I63060be3871456f2a5952c120820c3345ce8d268
Signed-off-by: nickboldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: make-release.sh [hunk 1]
```diff
@@ -65,14 +65,13 @@ fetchAndCheckout ()
   git fetch origin "${bBRANCH}:${bBRANCH}"; git checkout "${bBRANCH}"
 }
 
-# work in tmp dir if not already checked out by GH action
-if [[ ! -d ${REPO##*/} ]]; then
-  if [[ ! $TMP ]] || [[ ! -d $TMP ]]; then TMP=$(mktemp -d); fi
+# work in tmp dir if --use-tmp-dir (not required when running as GH action)
+if [[ $TMP ]] && [[ -d $TMP ]]; then
```
Comment (mkuznyetsov) [inline make-release.sh:69]: replace `[[ $TMP ]]` with `[[ -n $TMP ]]` ?

File: make-release.sh [hunk 2]
```diff
@@ -65,14 +65,13 @@ fetchAndCheckout ()
...
-if [[ ! -d ${REPO##*/} ]]; then
-  if [[ ! $TMP ]] || [[ ! -d $TMP ]]; then TMP=$(mktemp -d); fi
+# work in tmp dir if --use-tmp-dir (not required when running as GH action)
+if [[ $TMP ]] && [[ -d $TMP ]]; then 
   pushd "$TMP" > /dev/null || exit 1
   # get sources from ${BASEBRANCH} branch
   echo "Check out ${REPO} to ${TMP}/${REPO##*/}"
   git clone "${REPO}" -q
-  cd "${REPO##*/}" || exit 1
+  cd "${REPO##*/}" || true
```
Comment (mkuznyetsov) [inline make-release.sh:74]: is || true even needed?

File: make-release.sh (hunk 1 above)
Comment (nickboldt) [reply make-release.sh:69]:
```
$➔ if [[ -n $TMP ]]; then echo "TMP is set"; else echo "TMP not set"; fi
TMP not set
$➔ if [[ $TMP ]]; then echo "TMP is set"; else echo "TMP not set"; fi
TMP not set
```

same thing

**Review by mkuznyetsov — APPROVED** [review]

File: make-release.sh (hunk 2 above)
Comment (nickboldt) [reply make-release.sh:74]: not really but this way if the clone fails, the cd won't fail.

**Pull request merged**